scripts/amd/gemm/README.md

```diff
@@ -1,44 +1,164 @@
-# GEMM tuning script v2
+# GEMM tuning script (current v3.3)
 
-This is the v2 version of the gemm tuning script, which is based on @scxiao's v1 (https://github.com/ROCmSoftwarePlatform/triton/pull/309) and @alefimov-amd's thread pool https://github.com/ROCmSoftwarePlatform/triton/pull/310
+## matmul kernel
 
-### Main features
-- `rocprof` is used to measure the time for kernels in the full tuning space
-- Each kernel is executed 10 times and the execution time of the last instance is used
-- All kernels are compiled in parallel
-- Two modes for correctness checking
-    - During tuning, check correctness with the best perf_config for the current gemm size
-    - Without tuning, check correctness based on the tuning results, which includes best perf_config for each gemm size
-- The process takes about 30 - 40 minutes for the full tuning space with ~15000 configs
-- Limitations
-   - For now, only support fp16 as inputs. It should be trivial to extend to other types, but may require some work for mixed inputs
+The matmul kernel implementation can be found as [matmul_kernel.py](https://github.com/ROCm/triton/blob/triton-mlir/scripts/amd/gemm/matmul_kernel.py), which includes the following features:
+- grouping order of workgroup id, which is controled by `GROUP_SIZE_M`, that
+implements L2 cache optimization introduced in the [tutorial](https://triton-lang.org/main/getting-started/tutorials/03-matrix-multiplication.html#l2-cache-optimizations).
+- split-k algorithm, which is controled by `SPLIT_K`.
+- Bias along M dim, which is controled by `BIAS` and `bias_ptr`.
+- Masked load along K dim inside the loop, which is controled by `EVEN_K`.
+This means `BLOCK_SIZE_K` does not need to divide K dim.
 
-### Usage
-Go to the script dir
-```bash
-cd triton/scripts/amd/gemm/
+### Differences between the tutorial
+
+Unlike the [matmul tutorial](https://github.com/triton-lang/triton/blob/main/python/tutorials/03-matrix-multiplication.py) (referred as the tutorial), 
+the matmul kernel used in the tuning script (referred as the kerel) does not 
+guard load along M and N dim 
```

Comment:
I'm curious for this decision. Is it because we don't see enough shapes where M/N are not multiples of BLOCK_SIZE?